django/db/migrations/autodetector.py

```diff
@@ -576,6 +591,134 @@ def generate_renamed_models(self):
                             self.old_model_keys.add((app_label, model_name))
                             break
 
+    def generate_moved_models(self):
+        """
+        Find any moved models, generate the operations for them, and remove
+        the old entry from the model lists. Must be run before other
+        model-level generation.
+        """
+        added_models = self.new_model_keys - self.old_model_keys
+
+        for app_label, model_name in sorted(added_models):
+            model_state = self.to_state.models[app_label, model_name]
+            model_fields_def = self.only_relation_agnostic_fields(model_state.fields)
+            removed_models = self.old_model_keys - self.new_model_keys
+            for rem_app_label, rem_model_name in removed_models:
+                if rem_app_label != app_label:
+                    rem_model_state = self.from_state.models[
+                        rem_app_label, rem_model_name
+                    ]
+                    rem_model_fields_def = self.only_relation_agnostic_fields(
+                        rem_model_state.fields
+                    )
+
+                    if (
+                        model_fields_def == rem_model_fields_def
+                        and self.questioner.ask_move_model(
+                            rem_model_state,
+                            model_state,
+                        )
+                    ):
+                        db_table = model_state.options.get(
+                            "db_table",
+                        )
+                        if not db_table:
+                            self.questioner.ask_db_table_add(
+                                model_name, f"{rem_app_label}_{model_name}"
+                            )
+
+                        dependencies = []
+                        fields = list(model_state.fields.values()) + [
+                            field.remote_field
+                            for relations in self.to_state.relations[
+                                app_label, model_name
+                            ].values()
+                            for field in relations.values()
+                        ]
+                        for field in fields:
+                            if field.is_relation:
+                                dependencies.extend(
+                                    self._get_dependencies_for_foreign_key(
+                                        app_label,
+                                        model_name,
+                                        field,
+                                        self.to_state,
+                                    )
+                                )
+
+                        dependencies = list(dependencies) + [
+                            (rem_app_label, model_name, None, "alter")
+                        ]
```

Comment:
`dependencies` is already a list, isn't it?
```suggestion
                        dependencies.append(
                            (rem_app_label, model_name, None, "alter")
                        )
```